File: bigquery/src/test/java/com/dbx/agent/bigquery/BigQueryAgentMetadataTest.java
package com.dbx.agent.bigquery;

import com.dbx.agent.test.JdbcAgentFake;
import com.dbx.agent.test.JdbcMetadataSqlFake;
import java.sql.Connection;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class BigQueryAgentMetadataTest {
    @Test
    void quotesSchemaIdentifiersInMetadataSql() {
        BigQueryAgent agent = new BigQueryAgent();
        Connection fake = JdbcMetadataSqlFake.connection();
        JdbcAgentFake.setPrivateConnection(agent, fake);

        agent.listTables("bad`schema");
        agent.getColumns("bad`schema", "sample");

        Assertions.assertTrue(
            JdbcMetadataSqlFake.statements.stream()
                .anyMatch(statement -> statement.contains("FROM `bad``schema`.INFORMATION_SCHEMA.TABLES"))
        );
        Assertions.assertTrue(
            JdbcMetadataSqlFake.statements.stream()
                .anyMatch(statement -> statement.contains("FROM `bad``schema`.INFORMATION_SCHEMA.COLUMNS"))
        );
    }
}

File: bigquery/src/test/kotlin/com/dbx/agent/bigquery/BigQueryAgentMetadataTest.kt
package com.dbx.agent.bigquery

import com.dbx.agent.test.JdbcMetadataSqlFake
import com.dbx.agent.test.setPrivateConnection
import kotlin.test.Test
import kotlin.test.assertTrue

class BigQueryAgentMetadataTest {
    @Test
    fun `quotes schema identifiers in metadata SQL`() {
        val agent = BigQueryAgent()
        val fake = JdbcMetadataSqlFake.connection()
        setPrivateConnection(agent, fake)

        agent.listTables("bad`schema")
        agent.getColumns("bad`schema", "sample")

        assertTrue(
            JdbcMetadataSqlFake.statements.any {
                it.contains("FROM `bad``schema`.INFORMATION_SCHEMA.TABLES")
            }
        )
        assertTrue(
            JdbcMetadataSqlFake.statements.any {
                it.contains("FROM `bad``schema`.INFORMATION_SCHEMA.COLUMNS")
            }
        )
    }
}
